src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,46 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(animal) {
+    if (
+      animal instanceof Herbivore &&
+      !animal.hidden &&
+      !(animal instanceof Carnivore)
```

Comment:
[CODE LOGIC] - Since `Carnivore` also extends `Animal`, the check `!(animal instanceof Carnivore)` is unnecessary because `Herbivore` instances will never be an instance of `Carnivore`. You can safely remove this check.

Reply:
<img width="592" alt="image" src="https://github.com/user-attachments/assets/a8a3ba27-2a80-4f59-8384-c9bcafb396c0">